semgrep_output_v1.atd

```diff
@@ -554,52 +554,62 @@ type rule_times
   <ocaml attr="deriving show">
   <python decorator="dataclass(frozen=True)"> = {
   rule_id: rule_id;
+  (* TODO: time to parse what? *)
   parse_time: float;
   match_time: float;
 }
 
 (* ----------------------------- *)
 (* CLI *)
 (* ----------------------------- *)
-(* LATER: could merge with core_timing above
- * coupling: if you change the JSON schema below, you probably need to
+(* coupling: if you change the JSON schema below, you probably need to
  * also modify perf/run-benchmarks. Run locally  $ ./run-benchmarks --dummy --upload
  *)
-type cli_timing <ocaml attr="deriving show"> = {
-    rules: rule_id_dict list;
+type profile <ocaml attr="deriving show"> = {
+    (* List of rules, including the one read but not run on any target.
+     * This list is actually more an array which allows other
+     * fields to reference rule by number instead of rule_id
+     * (e.g., match_times further below) saving space in the JSON.
+     *
+     * Upgrade note: this used to be defined as a rule_id_dict where
+     * each rule_id was inside a {id: rule_id} record (maybe so
+     * we could have given more profiling info about each rule, but
+     * this was never used, so better to make it simple).
+     *)
+    rules: rule_id list;
+
+    (* LESS? could be part of profiling_times below instead *)
     rules_parse_time: float;
-    (* LATER: define a cli_profiling_times with more precise keys *)
+    (* coupling: semgrep_metrics.atd profilingTimes field?
+     * LATER? define a cli_profiling_times with more precise keys? 
+     * type cli_profiling_times <ocaml attr="deriving show"> = {
+     *   config_time: float;
+     *   core_time: float;
+     *   ignores_time: float;
+     *   total_time: float;
+     *  }
+     * LATER: get rid of profiler.dump_stats
+     *)
     profiling_times: (string * float) list
       <json repr="object">
       <python repr="dict">
       <ts repr="map">;
+
     targets: cli_target_times list;
     total_bytes: int;
 
+    (* maximum amount of memory used by Semgrep during its execution *)
     ?max_memory_bytes : int option;
   }
 
-(* LATER: should just use rule_id *)
-type rule_id_dict <ocaml attr="deriving show"> = {
-  id: rule_id;
-}
-
-(* LATER: get rid of profiler.dump_stats
-type cli_profiling_times <ocaml attr="deriving show"> = {
-   config_time: float;
-   core_time: float;
-   ignores_time: float;
-   total_time: float;
-}
-*)
-
-(* similar to the [target_time] and [rule_times] type above *)
 type cli_target_times <ocaml attr="deriving show"> = {
     path: fpath;
     num_bytes: int;
-    (* each elt in the list refers to a rule in cli_timing.rules *)
+    (* each elt in the list refers to a rule in profile.rules *)
     match_times: float list;
+    (* TODO: why a list? *)
```

Comment:
This doesn't need to be a list anymore. When we were first diagnosing performance, I recorded every time the file was read (including the later times that were just reloading the parsed file) to make sure reading the file wasn't taking a significant amount of time. Now that we know it isn't, we don't need to record this anymore